bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -738,9 +738,14 @@ private String updateStatusMessage(String message) {
         }
         var newBody = originalBody + "\n\n" + progressMarker + "\n" + message;
 
-        // TODO? Retrieve the body again here to lower the chance of concurrent updates
-        log.info("Updating PR body");
-        pr.setBody(newBody);
+        // Retrieve the body again here to lower the chance of concurrent updates
+        if (pr.body().equals(pr.latestBody())) {
+            log.info("Updating PR body");
+            pr.setBody(newBody);
+        } else {
+            log.info("PR body has been modified, won't update PR body this time");
+            return description;
```

Comment:
I thought the body will be updated in next round, so I didn't throw exception here.  Why is retry here better than updating body in next round?

Reply:
Hm, you are right, if the body has been updated since this CheckRun was triggered, we will get another one scheduled anyway. Since this isn't actually a failure condition, it's better to not throw exceptions that will annoy admins.